docs/mr_proposal.md

```diff
@@ -0,0 +1,187 @@
+# CUDA `<memory_resource>` Extension Proposal
+
+## Motivation
+
+Performance sensitive applications that make frequent dynamic memory allocations often find that allocating memory to be a significant overhead. 
+CUDA developers are even more acutely aware of the costs of dynamic allocation due to the relatively higher cost of `cudaMalloc/cudaFree` compared to standard `malloc/free`.
+As a result, developers devise custom, high-performance memory allocators as optimized as the application the allocator serves. 
+However, what works well for one application will not always satisfy another, which leads to a proliferation of custom allocator implementations. 
+Interoperation among these applications is difficult without an interface to enable sharing a common allocator.
+
+In Standard C++, [`Allocator`s](https://en.cppreference.com/w/cpp/named_req/Allocator) have traditionally provided this common interface.
+C++17 introduced [`<memory_resource>`](https://en.cppreference.com/w/cpp/header/memory_resource) and the [`std::pmr::memory_resource`](https://en.cppreference.com/w/cpp/memory/memory_resource) abstract class that defines a minimal interface for (de)allocating raw bytes and sits below `Allocator`s. 
+This polymorphic interface provides the lingua franca for those who trade in custom memory allocators. 
```

Comment:
I think the metaphor is unnecessary. :) "I'm a memory allocator merchant, sailing the high seas across the flat Earth."

```suggestion
This polymorphic interface provides a standard way to define, expose and share custom memory allocation. 
```

Reply:
I'm a sucker for a good metaphor :)